performance_test/helper_scripts/generate_apex_xml.py

```diff
@@ -0,0 +1,174 @@
+        required=False,
+        default='.'
+    )
+    args = parser.parse_args()
+    interface_whitelist = args.whitelist
+    xml_file = '{}/DEFAULT_FASTRTPS_PROFILES.xml'.format(args.directory)
+
+    doc = Document()
+    # Top level tag profiles
+    profiles = doc.createElement('profiles')
+
+    # Transport desciptors
+    transport_descriptors = doc.createElement('transport_descriptors')
+
+    # Transport desciptor
+    transport_descriptor = doc.createElement('transport_descriptor')
+    # Transport desciptor ID
+    transport_id = doc.createElement('transport_id')
+    transport_id.appendChild(doc.createTextNode('apex_transport_descriptor'))
+    transport_descriptor.appendChild(transport_id)
+    # Transport desciptor type
+    transport_type = doc.createElement('type')
+    transport_type.appendChild(doc.createTextNode('UDPv4'))
+    transport_descriptor.appendChild(transport_type)
+    # Transport desciptor whitelist
+    if interface_whitelist:
+        whitelists = doc.createElement('interfaceWhiteList')
+        for element in interface_whitelist:
+            address = doc.createElement('address')
+            address.appendChild(doc.createTextNode(element))
+            whitelists.appendChild(address)
+        transport_descriptor.appendChild(whitelists)
+    # Append transport descriptor to transport_descriptor
+    transport_descriptors.appendChild(transport_descriptor)
+    # Append transport descriptors to profile
+    profiles.appendChild(transport_descriptors)
+
+    # Participant
+    participant = doc.createElement('participant')
+    # Participant attributes
+    participant.setAttribute('profile_name', 'apex_profile')
+    participant.setAttribute('is_default_profile', 'true')
+    # Participant RTPS
+    rtps = doc.createElement('rtps')
+    # Participant RTPS name
+    name = doc.createElement('name')
+    name.appendChild(doc.createTextNode('Apex participant profile'))
+    rtps.appendChild(name)
+    # Participant RTPS useBuiltinTransports flag
+    use_btp = doc.createElement('useBuiltinTransports')
+    use_btp.appendChild(doc.createTextNode('false'))
+    rtps.appendChild(use_btp)
+    # Participant RTPS user transports
+    user_transports = doc.createElement('userTransports')
+    # Participant RTPS user transports ID
+    transport_id = doc.createElement('transport_id')
+    transport_id.appendChild(doc.createTextNode('apex_transport_descriptor'))
+    user_transports.appendChild(transport_id)
+    rtps.appendChild(user_transports)
+
+    # Append RTPS to participant
+    participant.appendChild(rtps)
+    # Append participant to profiles
+    profiles.appendChild(participant)
+    # Append profiles to document
+    doc.appendChild(profiles)
+
+    # Write to file
+    with open(xml_file, 'w') as f:
+        # Convert to string with 2-space identation,
+        # not taking the first line of the XML.
+        f.write(doc.toprettyxml(indent=' '*2).split('\n', 1)[1])
```

Comment:
The first line of the XML should read `<?xml version="1.0" encoding="UTF-8" ?>`, is there a reason why it should be removed? If not, I'd just output the XML document as is.

Reply:
Just for the shake of the XML being "cleaner". I just put it back